nifi/nifi-nar-bundles/nifi-kite-bundle/nifi-kite-processors/src/main/java/org/apache/nifi/processors/kite/AvroRecordConverter.java

```diff
@@ -0,0 +1,279 @@
+				}
+			case FLOAT:
+				try {
+					return Float.parseFloat(String.valueOf(content));
+				} catch (NumberFormatException e) {
+					throw new AvroConversionException("Cannot convert "
+							+ content + " to float");
+				}
+			default:
+				throw new AvroConversionException("Cannot convert to type "
+						+ nonNillOutput.getType());
+			}
+		}
+	}
+
+	/**
+	 * Get a mapping from output column to input column definition.
+	 * 
+	 * @param mappingStr
+	 * @return
+	 * @throws AvroConversionException
+	 */
+	private Map<String, String> getFieldMapping(JsonNode listNode) {
+		Map<String, String> result = Maps.newHashMap();
+		for (JsonNode mappingNode : listNode) {
+			result.put(mappingNode.get("target").getTextValue(), mappingNode
+					.get("source").getTextValue());
+		}
+
+		return result;
+	}
+
+	/**
+	 * Gets the field for a given name in a schema, handling . notation to step
+	 * into records.
+	 * 
+	 * @param fieldName
+	 * @param s
+	 * @return
+	 */
+	private static Field getFieldForName(String fieldName, Schema s) {
+		while (fieldName.contains(".")) {
+			// Recurse down the schema to find the right field.
+			int dotIndex = fieldName.indexOf('.');
+			String entityName = fieldName.substring(0, dotIndex);
+			s = undoNillableSchema(s);
+			s = s.getField(entityName).schema();
+			fieldName = fieldName.substring(dotIndex + 1);
+		}
+		s = undoNillableSchema(s);
+		return s.getField(fieldName);
+	}
+
+	protected Object getContentForName(Record entity, String fieldName, Schema s) {
+		IndexedRecord currentRecord = entity;
+		while (fieldName.contains(".")) {
+			// Recurse down the schema to find the right field.
+			int dotIndex = fieldName.indexOf('.');
+			String entityName = fieldName.substring(0, dotIndex);
+			Object innerRecord = currentRecord
+					.get(s.getField(entityName).pos());
+			// This may have been a null union type. Bail out now.
+			if (innerRecord == null) {
+				return null;
+			}
+			Preconditions.checkArgument(innerRecord instanceof IndexedRecord,
+					"Cannot access subfield " + fieldName
+							+ " because it is not a record");
+			currentRecord = (IndexedRecord) innerRecord;
+			s = undoNillableSchema(s);
+			s = s.getField(entityName).schema();
+			fieldName = fieldName.substring(dotIndex + 1);
+		}
+
+		s = undoNillableSchema(s);
+		return currentRecord.get(s.getField(fieldName).pos());
+	}
+
+	/**
+	 * If s is a union schema of some type with null, returns that type.
+	 * Otherwise just return schema itself.
+	 * 
+	 * Does not handle unions of schemas with anything except null and one type.
+	 * 
+	 * @param s
+	 * @return
+	 */
+	protected static Schema undoNillableSchema(Schema s) {
+		// Handle the case where s is a union type. Assert that this must be a
+		// union
+		// that only includes one record/map
+		if (s.getType() == Schema.Type.UNION) {
+			List<Schema> types = s.getTypes();
+			boolean foundOne = false;
+			Schema result = s;
+			for (Schema type : types) {
```

Comment:
I think it would be better to have different handling for optional fields and real unions with multiple possible types. Maybe the latter case is for a future version, but I think it still makes sense to limit this to two schemas: [ "null", ? ] and [ ?, "null" ].